django/core/cache/backends/memcached.py

```diff
@@ -214,3 +214,33 @@ def close(self, **kwargs):
         # libmemcached manages its own connections. Don't call disconnect_all()
         # as it resets the failover state and creates unnecessary reconnects.
         pass
+
+
+class PyMemcacheCache(BaseMemcachedCache):
+    """An implementation of a cache binding using pymemcache."""
+
+    def __init__(self, server, params):
+        import pymemcache.serde
+        super().__init__(server, params, library=pymemcache, value_not_found_exception=KeyError)
+        self._class = self._lib.HashClient
+        self._options = {
+            'allow_unicode_keys': True,
+            'default_noreply': False,
+            'serde': pymemcache.serde.pickle_serde,
+            **self._options,
+        }
+
+    @property
+    def client_servers(self):
+        output = []
+        for server in self._servers:
+            if server.startswith('unix:'):
+                output.append(server[5:])
+            elif server.startswith('/'):
+                output.append(server)
+            elif ':' not in server or server.endswith(']'):
+                output.append((server, 11211))
+            else:
+                host, port = server.rsplit(':', 1)
+                output.append((host, int(port)))
```

Comment:
Will we be able to remove this workaround after https://github.com/pinterest/pymemcache/pull/295?

Reply:
No, https://github.com/pinterest/pymemcache/pull/295 is only fixing it so that `HashClient` supports UNIX sockets properly, passing the paths to the underlying `Client` class that does support them. We are using `HashClient` as it supports multiple servers whereas the `Client` class does not.